andela/odin-ah-frontend
# #160452433 - Sign up page

#### What does this PR do?
- Added unit and integrated for SignUp Component
- Added validations on sign up form
- Consumed backend api for creating users
- Added Component for displaying Toast or Alert notifications
- Displays alert to the user when an error during registration
- Added Social signup button
- Added Verify Link page
- Added resend verification link page

#### Description of Task to be completed?
#### How should this be manually tested?
- Fetch and checkout to the branch
```$xslt
> git fetch
> git checkout ch-160452433-setup-signup-interface
> npm install 
```
- Rename the `.env.sample` file on the root directory of the cloned repo to `.env`. Uncomment and supply the following variable: `REACT_APP_HOST` which is the link to the frontend application; `REACT_APP_BASE_API_URL` which is the base URL for the backend api. For example, `https://odin-ah-backend-staging.herokuapp.com/api/v1`.
- To run the test first start the frontend server.

```$xslt 
> npm start
```
then run
```$xslt 
> npm run test
```
- Open http://localhost:3000 in your browser and click on the `Lunch Sign up box `
**Note**: Ensure to start the backend api server if `REACT_APP_BASE_API_URL` is pointing to your local machine.
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?

[#160452433](www.pivotaltracker.com/n/projects/2197379/stories/160452433)
#### Screenshots (if appropriate)

<img width="657" alt="screen shot 2018-11-10 at 9 22 48 am" src="https://user-images.githubusercontent.com/15909231/48299243-ce45ae80-e4ca-11e8-82e6-4216b4932261.png">

<img width="1368" alt="screen shot 2018-11-10 at 9 26 37 am" src="https://user-images.githubusercontent.com/15909231/48299239-cbe35480-e4ca-11e8-8bb9-98fc64ab8255.png">

<img width="740" alt="screen shot 2018-11-10 at 9 24 07 am" src="https://user-images.githubusercontent.com/15909231/48299241-cd148180-e4ca-11e8-860b-0c98562cbdf7.png">

#### Questions:
